src/SDL_log.c

```diff
@@ -422,6 +426,12 @@ SDL_LogOutput(void *userdata, int category, SDL_LogPriority priority,
         fprintf(pFile, "%s: %s\n", SDL_priority_prefixes[priority], message);
         fclose (pFile);
     }
+#elif defined(NXDK)
+    {
+        static char text[SDL_MAX_LOG_MESSAGE];
```

Comment:
Don't make this static, it'll waste a whole page and is not thread-safe.
Consider using `SDL_small_alloc` like the Windows code above does, it'll reduce the overhead and keep the risk of overflowing the stack minimal.